macros.in

```diff
@@ -1273,12 +1273,27 @@ end}
 # -S<scm name>	Sets the used patch application style, eg '-S git' enables
 #           	usage of git repository and per-patch commits.
 # -N		Disable automatic patch application
-# -p<num>	Use -p<num> for patch application	
+# -p<num>	Use -p<num> for patch application
 %autosetup(a:b:cDn:TvNS:p:)\
 %setup %{-a} %{-b} %{-c} %{-D} %{-n} %{-T} %{!-v:-q}\
 %{-S:%global __scm %{-S*}}\
 %{expand:%__scm_setup_%{__scm} %{!-v:-q}}\
 %{!-N:%autopatch %{-v} %{-p:-p%{-p*}}}
 
+# Macro to remove or replace tilde from version.
+# -v<version> Sets the version to be used for replacement. Default is %version.
+%version_notilde(v:) %{lua:
+    local sep = rpm.expand('%1')
+    local ver = rpm.expand('%{!-v:%{version}}%{-v:%{-v*}}')
```

Comment:
Another remark here is that you might want to use %{VERSION} as the default instead - that will always refer to the main package version, whereas %{version} will point to whatever the last Version: tag in the spec was (it's rare but sometimes sub-packages have their own versions).